data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaSourceCustomConsumer.java

```diff
@@ -21,34 +21,32 @@
 import org.slf4j.LoggerFactory;
 
 import java.time.Duration;
+import java.util.List;
 import java.util.Map;
 import java.util.HashMap;
 import java.util.Arrays;
 import java.util.ArrayList;
 import java.util.Collection;
-import java.util.List;
 import java.util.concurrent.atomic.AtomicBoolean;
 
-/**
- * * A utility class which will handle the core Kafka consumer operation.
- */
-public class KafkaSourceCustomConsumer implements ConsumerRebalanceListener {
+public class KafkaSourceCustomConsumer<T> implements ConsumerRebalanceListener {
```

Comment:
Also see if you can merge `MultithreadedConsumer` and this class by doing
`public class KafkaSourceCustomConsumer<T> implements Runnable, ConsumerRebalanceListener`

Reply:
Also, I don't think you need Generic type <T> here. You should be able to create consumer like this 
```
private KafkaConsumer consumer;
public KafkaSourceCustomConsumer(Properties consumerProperties,                                                                                    
                                     AtomicBoolean status,                                                                                             
                                     Buffer<Record<Event>> buffer,                                                                                     
                                     TopicConfig topicConfig,                                                                                          
                                     String schemaType,                                                                                                
                                     PluginMetrics pluginMetrics) { 
schema = MessageFormat.getByMessageFormatByName(schemaType);                                                                                   
        switch (schema) {                                                                                                                              
            case JSON:                                                                                                                                 
                this.consumer = new KafkaConsumer<String, JsonNode>(consumerProperties);                                                               
                break;                                                                                                                                 
            case AVRO:                                                                                                                                 
                this.consumer = new KafkaConsumer<String, GenericRecord>(consumerProperties);                                                          
                break;                                                                                                                                 
            case PLAINTEXT:                                                                                                                            
            default:                                                                                                                                   
                System.out.println("Creating plaintext consumer");                                                                                     
                this.consumer = new KafkaConsumer<String, String>(consumerProperties);                                                                 
                break;                                                                                                                                 
        }      
```

Reply:
We created a KafkaConsumer without generic type and made the changes accordingly.